vm/src/obj/objsequence.rs

```diff
@@ -137,6 +137,24 @@ impl<T: Clone> PySliceableSequence for Vec<T> {
     }
 }
 
+pub enum SequenceIndex {
```

Comment:
could we reuse `RangeIndex`? We'll need a "int-or-slice" typy for most `__getitem__`-like functions.

Reply:
actually, by the time I completed the rework, I realized that, yes, it potentially could have just used `RangeIndex` I only need to convert to `i32` once It came time to check bounds.
```rust
fn delindex(self, index: i32, vm: &VirtualMachine) -> PyResult {
        if let Some(pos_index) = self.get_pos(index) {
//...
}
```
could be
```rust
fn delindex(self, index: PyInt, vm: &VirtualMachine) -> PyResult {
        if let Some(pos_index) = self.get_pos(i32::try_from_object(vm, index.into_object())?) {
//...
}
```

however, I felt it would be best to leave it as is until directed to merge it. it seemed a little strange to be using something called `RangeIndex`.